packages/generator-single-spa/src/angular/generator-single-spa-angular.js

```diff
@@ -35,12 +81,51 @@ module.exports = class SingleSpaAngularGenerator extends Generator {
     } else {
       spawnSync(command, args.concat(["add", "single-spa-angular"]), {
         stdio: "inherit",
-        cwd
+        cwd: this.cwd,
       });
 
       console.log(
         "For further routing setup, see https://single-spa.js.org/docs/ecosystem-angular#configure-routes"
       );
     }
   }
+
+  // Including single-spa as a dependency has not been fully resolved.
+  // It is now included in the Angular schematic https://github.com/single-spa/single-spa-angular/pull/193
+  // but that has only been released as an alpha version https://github.com/single-spa/single-spa-angular/releases/tag/v4.1.0-alpha.0
+  async validateSingleSpaDependency() {
+    const { dependencies } = await this.fs.readJSON(`${this.cwd}/package.json`);
+    if (dependencies && !dependencies["single-spa"]) {
+      const install = (cmd, opts) =>
+        spawnSync(cmd, opts, {
+          stdio: "inherit",
+          cwd: this.cwd,
+        });
+      switch (this.options.packageManager) {
+        case "npm":
+          install("npm", ["install", "single-spa"]);
+          break;
+        case "yarn":
+          install("yarn", ["add", "single-spa"]);
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  async finished() {
+    console.log(
+      chalk.bgWhite.black(
+        `Project setup complete!
+Steps to test your Angular single-spa application:
+1. Run '${this.options.packageManager}${
+          this.options.packageManager === "npm" ? " run" : ""
+        } serve:single-spa:${this.options.projectName}'
+2. Go to http://single-spa-playground.org/playground/instant-test?name=${
+          this.options.projectName
+        }&url=%2F%2Flocalhost%3A4200%2Fmain.js&framework=angular to see it working!`
```

Comment:
single-spa-playground uses the framework check to determine whether to include ZoneJS on the page. I think it requires a full page reload when adding Angular for the first time so that ZoneJS gets included. Have you seen whether that works when testing it on the playground?

Reply:
I did but by running the playground locally. Now with https://github.com/single-spa/single-spa-playground/pull/27 being merged, I can test in the prod one. You're right though. I think I was manually refreshing without thinking about it.